scripts/translation/lib/RevtagParser.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ *  +----------------------------------------------------------------------+
+ *  | Copyright (c) 1997-2023 The PHP Group                                |
+ *  +----------------------------------------------------------------------+
+ *  | This source file is subject to version 3.01 of the PHP license,      |
+ *  | that is bundled with this package in the file LICENSE, and is        |
+ *  | available through the world-wide-web at the following url:           |
+ *  | https://www.php.net/license/3_01.txt.                                |
+ *  | If you did not receive a copy of the PHP license and are unable to   |
+ *  | obtain it through the world-wide-web, please send a note to          |
+ *  | license@php.net, so we can mail you a copy immediately.              |
+ *  +----------------------------------------------------------------------+
+ *  | Authors:     André L F S Bacci <ae php.net>                          |
+ *  +----------------------------------------------------------------------+
+ *  | Description: Parse revision and credits from XML comments.           |
+ *  +----------------------------------------------------------------------+
+ */
+
+require_once __DIR__ . '/all.php';
+
+class RevtagInfo
+{
+    public string $revision = "";
+    public string $maintainer = "";
+    public string $status = "";
+    public string $credits = "";
+    public array  $errors = [];
+}
+
+class RevtagParser
+{
+    static function parseInto( string $lang , RevcheckFileList & $list )
+    {
+        foreach( $list->list as $entry )
+            $entry->revtag = RevtagParser::parseFile( $lang . '/' . $entry->file );
+    }
+
+    public static function parseFile( string $filename ): RevtagInfo|null
+    {
+        $doc = XmlUtil::loadFile( $filename );
+        $ret = new RevtagInfo;
+        RevtagParser::parseNodeRecurse( $doc , $ret , $filename );
+        return $ret;
+    }
+
+    public static function parseText( string $contents ): RevtagInfo|null
```

Comment:
How can this function return `null`?

Reply:
In general, there is no grantee that a particular dictionary will contain all the possible keys.

In specific, file paths are used as keys, and is very common for translations to don't have all files translated. Then, there will be keys on `en` list that does not exists in $LANG list;